sdk/src/account_meta.rs

```diff
@@ -0,0 +1,119 @@
+//! Types used
+
+use anchor_lang::{AnchorDeserialize, AnchorSerialize};
+use solana_program::pubkey::Pubkey;
+
+use crate::{
+    compressed_account::CompressedAccountWithMerkleContext,
+    merkle_context::{
+        pack_address_merkle_context, pack_merkle_context, AddressMerkleContext,
+        PackedAddressMerkleContext, PackedMerkleContext, RemainingAccounts,
+    },
+};
+
+#[derive(Debug, Clone, AnchorSerialize, AnchorDeserialize, PartialEq, Default)]
+pub struct LightAccountMeta {
+    /// Lamports.
+    pub lamports: Option<u64>,
+    /// Address of the account (the address can change).
+    pub address: Option<[u8; 32]>,
+    /// Data of the account.
+    pub data: Option<Vec<u8>>,
+    /// Merkle tree.
+    pub merkle_context: Option<PackedMerkleContext>,
+    /// Merkle tree root index.
+    pub merkle_tree_root_index: Option<u16>,
+    /// Output Merkle tree.
+    pub output_merkle_tree_index: Option<u8>,
+    /// Address Merkle tree. Set only when adding or updating the address.
+    pub address_merkle_context: Option<PackedAddressMerkleContext>,
+    /// Address Merkle tree root index. Set only when adding or updating the
+    /// address.
+    pub address_merkle_tree_root_index: Option<u16>,
+}
+
+impl LightAccountMeta {
+    #[allow(clippy::too_many_arguments)]
+    pub fn new_init(
+        lamports: Option<u64>,
+        data: Option<Vec<u8>>,
+        output_merkle_tree: &Pubkey,
+        address_merkle_context: Option<&AddressMerkleContext>,
+        address_merkle_tree_root_index: Option<u16>,
+        remaining_accounts: &mut RemainingAccounts,
+    ) -> Self {
+        let output_merkle_tree_index = remaining_accounts.insert_or_get(*output_merkle_tree);
+        let address_merkle_context =
+            address_merkle_context.map(|ctx| pack_address_merkle_context(ctx, remaining_accounts));
+        Self {
+            lamports,
+            address: None,
+            data,
+            merkle_context: None,
+            merkle_tree_root_index: None,
+            output_merkle_tree_index: Some(output_merkle_tree_index),
+            address_merkle_context,
+            address_merkle_tree_root_index,
+        }
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    pub fn new_mut(
+        compressed_account: &CompressedAccountWithMerkleContext,
+        lamports: Option<u64>,
+        merkle_tree_root_index: u16,
+        output_merkle_tree_index: Option<u8>,
+        address_merkle_context: Option<&AddressMerkleContext>,
+        address_merkle_tree_root_index: Option<u16>,
```

Comment:
imo we can omit these fields, since the design pattern I expect creates an address at account creation not when mutating an account and we pass a lot of None values at initialization.
What about removing these input fields and adding a `create_new_address` to `LightAccountMeta` method which allows to specify a new account explicitly if required?
(This would also make it intuitive for devs that to keep the existing address they don't need to specify these inputs.)

Reply:
agreed on removing if possible. create_new_address for existing account is such an anti-pattern i dont even think it's necessary to add that helper right now! what do you think @vadorovsky